core/block_processor.go

```diff
@@ -56,16 +56,31 @@ type BlockProcessor struct {
 	eventMux *event.TypeMux
 }
 
-// TODO: type GasPool big.Int
-//
-// GasPool is implemented by state.StateObject. This is a historical
-// coincidence. Gas tracking should move out of StateObject.
-
 // GasPool tracks the amount of gas available during
 // execution of the transactions in a block.
-type GasPool interface {
-	AddGas(gas, price *big.Int)
-	SubGas(gas, price *big.Int) error
+// The zero value is a pool with zero gas available.
+type GasPool struct{ bal *big.Int }
```

Comment:
This value ends up as the gasUsed field in the block header, if named so we're closer to formal spec in naming, which always helps (new readers).